### PR TITLE
feat(datasets): streaming read-back + HF Storage Bucket sync (physical-AI data loop)

### PR DESCRIPTION
**Pull request by cagataycali**
## Summary

Integrates the LeRobot **streaming data loop** and **Hugging Face Storage Buckets** into strands-robots dataset collection — closing the physical-AI data loop: **collect → bucket → stream-to-train (no download) → deploy** back to hardware.

Today strands-robots can record a `LeRobotDataset` (sim + hardware) and `push_to_hub()` to a versioned repo, but it has **no streaming read-back** and **no bucket-write** path. This PR adds both, natively.

> Single clean commit on top of `main`. Supersedes #606 (which carried merge noise from concurrent main updates).

### Native API (no library internals leaking into user code)
- **`Robot().stream_dataset(repo_id, **kw)`** — in-process streaming read-back, the counterpart to `start_recording`/`stop_recording`. The same `Robot()` that records can stream; camera frames decode on the fly from the MP4 shards.
- **`stop_recording(..., bucket=, run_id=)`** — one kwarg syncs the finalized episode into a mutable, Xet-deduped HF Storage Bucket (`hf://buckets/...`). Finalizes first so `meta/` (stats/info) ships for downstream normalization.
- **`DatasetRecorder.sync_to_bucket(bucket, run_id=None)`** — the underlying sync via the `hf` CLI (graceful error if CLI/login missing).
- **`StreamingDatasetReader`** (`strands_robots.streaming_dataset`) — version-tolerant wrapper over lerobot's `StreamingLeRobotDataset`: lazy import, `**kwargs` introspection, delta-grid validation parity, and a `drop_videos` proprio-only path for edge devices without a torchcodec wheel. Lazily exported from the package root.

### Dependencies (the `[lerobot]` extra is now self-sufficient for the data loop)
- `lerobot[feetech,dataset]` — pulls `datasets` + `pandas` + `pyarrow` + `av` + `torchcodec` so streaming video decode works out of the box.
- `torchcodec` pinned to track the resolved torch C++ ABI: `0.10` on darwin/arm64 (torch 2.10), `0.11` on linux/aarch64 (torch 2.11). Fixes the `dlopen "Symbol not found: c10::MessageLogger"` mismatch that 0.7/0.8 hit against torch 2.10.
- `test_customer_workflow_regressions` updated to assert `feetech` is a **member** of the lerobot extras set (so `lerobot[feetech,dataset]` passes) instead of matching the brittle literal `lerobot[feetech]`.

### macOS zero-touch dyld shim (`strands_robots/_dyld.py`)
torchcodec links ffmpeg via `@rpath`, but Homebrew's ffmpeg (`/opt/homebrew/lib`) isn't on the default dyld search path → `import torchcodec` fails with `Library not loaded: @rpath/libavutil.59.dylib`. `os.environ` mid-process can't fix it (dyld snapshots `DYLD_*` at launch) and `ctypes` preload doesn't satisfy `@rpath` on macOS. So `ensure_ffmpeg_on_dyld_path()` runs eagerly at `import strands_robots`:
- No-op unless macOS + Homebrew ffmpeg present + torchcodec installed + var not already set.
- Exports `DYLD_FALLBACK_LIBRARY_PATH` so **child processes** (DataLoader workers, subprocess training) inherit it.
- **Re-execs the interpreter once** (guarded against loops) for the current process — but **only when safe**: refuses inside Jupyter / IPython / pytest / REPL / `python -c`, where it warns with the one-line `export` instead.
- Opt out with `STRANDS_ROBOTS_NO_DYLD_SHIM=1`.

### Example + docs
- `examples/06_agent_collect_and_stream.py` — a Strands `Agent` records a dataset from one NL prompt, then `sim.stream_dataset()` reads it back with full camera decode (~50 lines, no torchcodec/av plumbing).
- README: new **"Recording & streaming datasets"** section + the macOS note + `STRANDS_ROBOTS_NO_DYLD_SHIM` env var.
- `docs/recording.md`: `sync_to_bucket` in the methods table + a **"Stream back (no full download)"** section.

## Verification (local, uv + py3.12, lerobot 0.5.2, torch 2.10 + torchcodec 0.10, `MUJOCO_GL=cgl`)
- Agent-driven sim recording: 60 frames / 1 episode, 2 cameras, `libsvtav1`.
- `sim.stream_dataset()` yields `state(6,)` + `action(6,)` + **decoded** camera tensors `(3, 480, 640)` straight from MP4 shards — zero re-download.
- Zero-touch confirmed: `env -u DYLD_FALLBACK_LIBRARY_PATH python examples/06_*.py` streams video with no manual setup.
- **86 tests pass** (streaming/shim/facade + recorder + customer-workflow). pytest correctly does **not** re-exec.

## Notes / scope
- For full **training**, the upstream trainer already uses the same engine via `python -m lerobot.scripts.train dataset.streaming=true` — no new code needed there; `StreamingDatasetReader`'s home is in-process eval/replay/inspection.
- True `hf://buckets/...` **streaming** (vs collection-time sync) needs a small upstream lerobot change to `LeRobotDatasetMetadata.url_root`; the recommended path today is collect→bucket + stream from a published dataset repo (near-instant server-side `hf buckets cp`). Tracked as a follow-up.
- Backward compatible: all additions; `stop_recording()`'s new args are keyword-only with prior defaults.

## Files (11)
```
 strands_robots/streaming_dataset.py            +193  (new)
 strands_robots/_dyld.py                         +159  (new)
 strands_robots/dataset_recorder.py             +107  (sync_to_bucket + push_to_hub kwargs)
 strands_robots/simulation/mujoco/recording.py   +81  (stream_dataset, bucket=)
 strands_robots/__init__.py                       +16  (lazy export + shim)
 pyproject.toml                                   +23  (lerobot[dataset] + torchcodec)
 examples/06_agent_collect_and_stream.py          +64  (new)
 tests/test_streaming_dataset.py                 +335  (new)
 tests/test_customer_workflow_regressions.py      +16  (feetech-as-member assertion)
 README.md                                        +63
 docs/recording.md                                +54
```

🤖 Authored with assistance from DevDuck.

### CODE REVIEW

**Review by yinsong1986 — APPROVED** [review]
## Summary

Adds the LeRobot streaming read-back path (`Robot().stream_dataset()` / `StreamingDatasetReader`) and an HF Storage Bucket sync (`DatasetRecorder.sync_to_bucket()`, surfaced via `stop_recording(bucket=, run_id=)`) to close the collect -> bucket -> stream -> deploy data loop, plus a macOS dyld shim so torchcodec finds Homebrew ffmpeg with zero setup. All changes are additive; `stop_recording`'s new args are keyword-only with prior defaults.

## What's good

- The agent-reachable subprocess surface is properly hardened: `bucket`/`run_id` are validated by leading-alphanumeric-anchored allowlist regexes that reject `..` traversal, shell metacharacters, and `-flag` argv option-injection; the default `run_id` derived from `repo_id` is re-validated; all `hf` calls are argv-style and gated on `shutil.which("hf")`. Rejection paths are pinned by parametrized tests (`test_sync_to_bucket_rejects_unsafe_{bucket,run_id}`).
- AGENTS.md compliance: error paths return `{"status": "error", ...}` dicts rather than raising; `finalize()` runs before bucket sync so `meta/` ships; the new `STRANDS_ROBOTS_NO_DYLD_SHIM` env var is documented in the README Configuration table; non-ASCII is confined to comments/docstrings (user-facing tool-result and log strings stay ASCII).
- Scope discipline: the dyld shim is a hard no-op off macOS and the streaming reader's kwarg forwarding tolerates a narrow constructor (tested), so neither path regresses Linux/Jetson or the existing recording flow.

## Verification suggestions

- Standard CI (`hatch run test` / `lint`) covers the changed paths; `pytest tests/test_streaming_dataset.py` passes 27/27 locally without lerobot or the `hf` CLI installed.
